### PR TITLE
Add whitelist for mainnet

### PR DESCRIPTION
**Pull request by alecande11**
Implemented some fixes from PR #118 by @Jared-TFL:
- fix error that prevent compiling
- hadle testnet
- use same `proposals.json` file with this format:
```js
{
  mainnet: number[]
  classic: number[]
}
```

It will need another PR on terra-money/asset to update proposals.json.

### CODE REVIEW

File: src/pages/gov/ProposalsByStatus.tsx [hunk 1]
```diff
@@ -15,13 +15,15 @@ import styles from "./ProposalsByStatus.module.scss"
...
-  const [showAll, setShowAll] = useState(!isClassic)
+  const [showAll, setShowAll] = useState(!isClassic && !isMainnet)
   const toggle = () => setShowAll((state) => !state)
 
-  const { data: whitelist, ...whitelistState } = useTerraAssets<number[]>(
-    "/station/proposals.json"
-  )
+  const { data: whitelist, ...whitelistState } = useTerraAssets<{
+    classic: number[]
+    mainnet: number[]
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]:
```
{ [name: string]?: number[] }
```

File: src/pages/gov/ProposalsByStatus.tsx [hunk 2]
```diff
@@ -33,7 +35,7 @@ const ProposalsByStatus = ({ status }: { status: Proposal.Status }) => {
 
     const proposals =
       status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && !showAll
-        ? data.filter(({ id }) => whitelist.includes(id))
+        ? data.filter(({ id }) => (whitelist[isClassic ? "classic" : "mainnet"] || []).includes(id))
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]: `whitelist[networkName]`

File: src/pages/gov/ProposalsByStatus.tsx [hunk 3]
```diff
@@ -69,7 +67,7 @@ const ProposalsByStatus = ({ status }: { status: Proposal.Status }) => {
   return (
     <Fetching {...state}>
       <Col>
-        {isClassic && status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && (
+        {(isClassic || isMainnet) && status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && (
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]: `(whitelistExists) && ...`

Comment (simcheolhwan) [reply src/pages/gov/ProposalsByStatus.tsx]: `const whitelistExists = !!whitelist[networkName]`

File: src/pages/gov/ProposalsByStatus.tsx (hunk 2 above)
Comment (simcheolhwan) [reply src/pages/gov/ProposalsByStatus.tsx]: (it may be undefined)

**Review by simcheolhwan — CHANGES_REQUESTED** [review]

File: src/data/query.ts [hunk 4]
```diff
@@ -29,9 +29,7 @@ export const combineState = (...results: QueryState[]) => ({
 /* queryKey */
 const mirror = <T>(obj: T, parentKey?: string): T =>
   Object.entries(obj).reduce((acc, [key, value]) => {
-    const next = value
-      ? mirror(value, key)
-      : [parentKey, key].filter(Boolean).join(".")
+    const next = value ? mirror(value, key) : [parentKey, key].filter(Boolean).join(".")
```
Comment (simcheolhwan) [inline src/data/query.ts]: Revert this please

File: src/pages/gov/ProposalsByStatus.tsx [hunk 5]
```diff
@@ -11,17 +11,18 @@ import { Toggle } from "components/form"
...
 import styles from "./ProposalsByStatus.module.scss"
+import { useNetworkName } from "data/wallet"
 
 const ProposalsByStatus = ({ status }: { status: Proposal.Status }) => {
   const { t } = useTranslation()
-  const isClassic = useIsClassic()
+  const networkName = useNetworkName()
 
-  const [showAll, setShowAll] = useState(!isClassic)
+  const { data: whitelist, ...whitelistState } = useTerraAssets<{ [key: string]: number[] }> ("/station/proposals.json")
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]: ` { data, ... ` instead of `{ data: whitelist, ...` and

File: src/pages/gov/ProposalsByStatus.tsx [hunk 6]
```diff
@@ -11,17 +11,18 @@ import { Toggle } from "components/form"
...
+
+  const whitelistExists = whitelist && whitelist[networkName]
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]:
```
const whitelist = data?.[networkName]
```

then

File: src/pages/gov/ProposalsByStatus.tsx [hunk 7]
```diff
@@ -33,7 +34,7 @@ const ProposalsByStatus = ({ status }: { status: Proposal.Status }) => {
 
     const proposals =
       status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && !showAll
-        ? data.filter(({ id }) => whitelist.includes(id))
+        ? data.filter(({ id }) => (whitelist[networkName] || []).includes(id))
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]: `whitelist?.includes(id)`

File: src/pages/gov/ProposalsByStatus.tsx [hunk 8]
```diff
@@ -69,7 +66,7 @@ const ProposalsByStatus = ({ status }: { status: Proposal.Status }) => {
   return (
     <Fetching {...state}>
       <Col>
-        {isClassic && status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && (
+        {whitelistExists && status === Proposal.Status.PROPOSAL_STATUS_VOTING_PERIOD && (
```
Comment (simcheolhwan) [inline src/pages/gov/ProposalsByStatus.tsx]: `!!whitelist &&`

**Review by simcheolhwan — CHANGES_REQUESTED** [review]

File: src/pages/gov/ProposalsByStatus.tsx (hunk 5 above)
Comment (alecande11) [reply src/pages/gov/ProposalsByStatus.tsx]: data is already defined below, I renamed it to whitelistData and applied all the other changes

**Review by simcheolhwan — APPROVED** [review]

**Pull request merged**